packages/modal/src/ModalProvider.tsx

```diff
@@ -0,0 +1,9 @@
+import { ZIndexManager } from '@xcritical/utils';
+
+
+const { Provider: ModalProvider, ZIndexContext } = ZIndexManager();
```

Comment:
I think this file is unnecessary. You can just use **import { ZIndexManager } from '@xcritical/utils';** inside packages/modal/src/Modal.tsx